src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+                    f32 yOffset;
+                    f32 xScale;
+
+                    xScale = sKoumePtr->actor.scale.x * 3000.0f;
+                    yOffset = Rand_CenteredFloat(xScale * 2.0f);
+                    pos.x = 3000.0f;
+                    pos.y = 400.0f + yOffset;
+                    pos.z = 0.0f;
+                    BossTw_AddMergeFlameEffect(globalCtx, &pos, Rand_ZeroFloat(5.0f) + 10.0f,
+                                               sqrtf(SQ(xScale) - SQ(yOffset)), Rand_ZeroFloat(1.99f));
+                }
+
+                if (sKoumePtr->actor.scale.x <= 0.0051f) {
+                    Vec3f pos;
+                    Vec3f velocity;
+                    Vec3f accel;
+
+                    this->actor.world.pos.y = 400.0f;
+
+                    for (i = 0; i < 50; i++) {
+                        pos = this->actor.world.pos;
+                        velocity.x = Rand_CenteredFloat(20.0f);
+                        velocity.y = Rand_CenteredFloat(20.0f);
+                        velocity.z = Rand_CenteredFloat(20.0f);
+                        pos.x += velocity.x;
+                        pos.y += velocity.y;
+                        pos.z += velocity.z;
+                        accel.z = accel.y = accel.x = 0.0f;
+                        BossTw_AddFlameEffect(globalCtx, &pos, &velocity, &accel, Rand_ZeroFloat(10.0f) + 25.0f,
+                                              velocity.x < 0.0f);
+                    }
+
+                    this->csState1 = 1;
+                    this->visible = true;
+                    this->actor.flags |= 1;
+                    this->actor.shape.rot.y = 0;
+                    BossTw_SetupWait(sKotakePtr, globalCtx);
+                    BossTw_SetupWait(sKoumePtr, globalCtx);
+                    Actor_SetScale(&this->actor, 0.0f);
+                    Animation_MorphToPlayOnce(&this->skelAnime, &D_06038E2C, 0.0f);
+                    this->workf[ANIM_SW_TGT] = Animation_GetLastFrame(&D_06038E2C);
+                    this->timers[0] = 50;
+                    func_8002DF54(globalCtx, &this->actor, 2);
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_TWINROBA_TRANSFORM);
+                    Audio_QueueSeqCmd(0x1B);
+                }
+            }
+
+            sKotakePtr->actor.scale.x = sKotakePtr->actor.scale.y = sKotakePtr->actor.scale.z =
+                sKoumePtr->actor.scale.y = sKoumePtr->actor.scale.z = sKoumePtr->actor.scale.x;
+            break;
+
+        case 1:
+            if (Animation_OnFrame(&this->skelAnime, this->workf[ANIM_SW_TGT])) {
+                Animation_MorphToLoop(&this->skelAnime, &D_06032BF8, -15.0f);
+            }
+
+            sEnvType = -1;
+            globalCtx->envCtx.unk_BD = 4;
+            Math_ApproachF(&globalCtx->envCtx.unk_D8, 1, 1, 0.1f);
+            // fallthrough
+        case 2:
+            SkelAnime_Update(&this->skelAnime);
+            Math_ApproachF(&this->actor.scale.x, 0.0069999993f, 1, 0.0006999999f);
```

Comment:
Most of these off-by-ones are probably `0.01f` times the corresponding float or double. Also, for consistency,
```suggestion
            Math_ApproachF(&this->actor.scale.x, 0.006999999f, 1, 0.0006999999f);
```

Reply:
did not match